docs/examples.md

```diff
@@ -164,9 +164,149 @@ object Main extends CommandIOApp("mkString", "Concatenates strings from stdin")
 
 @:@
 
+## Parsing and transforming a CSV file
+
+Here, [fs2-data-csv] is used to read and parse a comma separated file. 
+Manual encoders and decoders are defined for our `Passenger`s to show you how to do everything from scratch.
+
+Let's start with a CSV file that has records of fictious passengers registered for a flight:
+
+```
+id,First Name,Age,flight number,destination
+1,Seyton,44,WX122,Tanzania
+2,Lina,,UX199,Greenland
+3,Grogu,,SW999,Singapore
+```
+
+
+@:select(scala-version)
+
+@:choice(scala-3)
+```scala mdoc:reset:silent
+//> using lib "org.typelevel::toolkit::@VERSION@"
+
+import cats.effect.*
+import fs2.text
+import fs2.data.csv.*
+import fs2.data.csv.generic.semiauto._
+import fs2.io.file.{Path, Flags, Files}
+import cats.data.NonEmptyList
+
+case class Passenger(
+    id: Long,
+    firstName: String,
+    age: Either[String, Int],
+    flightNumber: String,
+    destination: String
+)
+
+object Passenger:
+  // Here we define a manual decoder for each row in our CSV
+  given csvRowDecoder: CsvRowDecoder[Passenger, String] with
+    def apply(row: CsvRow[String]): DecoderResult[Passenger] =
+      for
+        id <- row.as[Long]("id")
+        firstName <- row.as[String]("First Name")
+        ageOpt <- row.asNonEmpty[Int]("Age")
+        flightNumber <- row.as[String]("flight number")
+        destination <- row.as[String]("destination")
+      yield
+        val age = ageOpt.toRight[String]("N/A")
+        Passenger(id, firstName, age, flightNumber, destination)
+
+    // Here we define a manual encoder for encoding Passenger classes to a CSV
+    given csvRowEncoder: CsvRowEncoder[Passenger, String] with
+      def apply(p: Passenger): CsvRow[String] =
+        CsvRow.fromNelHeaders(
+          NonEmptyList.of(
+            (p.firstName, "first_name"),
+            (p.age.toString(), "age"),
+            (p.flightNumber, "flight_number"),
+            (p.destination, "destination")
+          )
+        )
+
+val input = Files[IO]
+  .readAll(Path("./example.csv"), 1024, Flags.Read)
+  .through(text.utf8.decode)
+  .through(decodeUsingHeaders[Passenger]())
+
+object CSVPrinter extends IOApp.Simple:
+  val run: IO[Unit] =
+    input.evalTap(p => IO.println(p)).compile.drain
+```
+
+
+@:choice(scala-2)
+```scala mdoc:reset:silent
+
+//> using lib "org.typelevel::toolkit::0.0.2"
+//> using lib "org.gnieh::fs2-data-csv::1.6.1"
+//> using lib "org.gnieh::fs2-data-csv-generic::1.6.1"
+
+import cats.effect.*
+import fs2.text
+import fs2.data.csv.*
+import fs2.data.csv.generic.semiauto._
+import fs2.io.file.{Path, Flags, Files}
+import cats.data.NonEmptyList
+
+case class Passenger(
+    id: Long,
+    firstName: String,
+    age: Either[String, Int],
+    flightNumber: String,
+    destination: String
```

Comment:
I was thinking of making `destination` or `flightNumber` an `opaque` type in Scala, and show off newtypes-ish usage in Scala. Is this a bit overkill for an example?
Ideally we would have 2 examples, one that shows off `deriveCsv...` derivations and one that shows off manual codecs.

Reply:
I'll answer you with a question. Is showing this difference worth a different example? We may increase the number of examples this way.
Another chance is creating a separate markdown document that contains examples specific to fs2 or fs2-csv+generic. WDYT?

Reply:
That is a good question, I think showing off opaque types in these examples is more of a flex than anything, because I don't know many people that would type out almost everything for a script, even if it's a good practice.

As for a separate markdown document that showing off `fs2-data-csv-generic` I think it's best to put that for a later time, because the `fs2-data` docs give a good example on how to use automatic derivation.

Reply:
Okay, that might be a good idea. In the future we may a have, let's say, a different set of examples for fs2, one for http4s, one for decline and so on. I have the fear that they might be somehow mocking each library's documentation, but that's solvable adding examples that are more "problem-solving oriented".
For what concerns this PR, I think that the example is fine :)

Reply:
> I have the fear that they might be somehow mocking each library's documentation

One advantage we have here, that other projects docs do not, is we can actually use multiple libraries together :) e.g. you can mix http4s + fs2-data + decline. So we should flex examples like that.

Reply:
That's a good point, let's add a few examples and let's see how to structure them in the doc :)

Reply:
I think sticking to the "problem-solving oriented" examples is the best here. It'll need to show off how to do borin' ol' programmer things using Typelevel in a beginner-friendly way.

@armanbilge beat me to it, but I agree that we want to have examples that show that off.
We could have something that gets pokemon from a pokemon api and writes it to a csv file. Or a commandline app that gets data from an API and outputs JSON, with an unit test.

These examples probably wouldn't fit in the examples page :P

Reply:
Well, how about a stuffed example that:
- accepts from command line 2 inputs: a pokemon kind and generation (defaulted to 1)
- gets all pokemons of that type in that generation
- writes a csv with all results (restricted to a few properties)

Will it be too long?